### PR TITLE
Separate workflows for webapp and query-API deployment in CircleCI

### PR DESCRIPTION
**Pull request by pylipp**
Please see the attached Trello card for an introduction.

I also updated the [deployment guide](https://docs.google.com/document/d/1zfqQ4yRUf4C-fbhMWfzKzBWPoDAvt4tF8nVI8fIrQrY/edit#) where needed.

@haguesto before merging this, I'd like to do a production deploy of v2 (to have established changes published first; then deploy again with the new CircleCI config which can result in some extra work for fixes).

Another thing we could consider is having a dedicated sentry project for the API.

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 Separate workflows for webapp and query-API deployment in CircleCI](https://trello.com/c/diQCkS58/971-20-separate-workflows-for-webapp-and-query-api-deployment-in-circleci)

**Review by HaGuesto — APPROVED** [review]
from the code review it is approved. Let's do the deploy together @pylipp

File: back/boxtribute_server/app.py [hunk 1]
```diff
@@ -20,3 +24,25 @@ def configure_app(app, *blueprints, database_interface=None, **mysql_kwargs):
...
     app.config["DATABASE"] = database_interface or create_db_interface(**mysql_kwargs)
     db.init_app(app)
+
+
+def main(*blueprints):
+    """Integrate Sentry SDK. Create and configure Flask app."""
+    # dsn/environment/release: reading SENTRY_* environment variables set in CircleCI
+    sentry_sdk.init(
+        integrations=[FlaskIntegration()],
+        traces_sample_rate=float(os.getenv("SENTRY_TRACES_SAMPLE_RATE", 1.0)),
```
Comment (HaGuesto) [inline back/boxtribute_server/app.py:34]: What is this for?

Comment (pylipp) [reply back/boxtribute_server/app.py:34]: Cf. https://docs.sentry.io/platforms/python/configuration/sampling/#setting-a-uniform-sample-rate

**Pull request merged**